src/backend/executor/nodePhysicalGroupLookup.c

```diff
@@ -45,24 +48,89 @@ ExecInitPhysicalGroupLookup(PhysicalGroupLookup *node, EState *estate, int eflag
 }
 
 /*
- * ExecExtractPhysical
+ * ExecPhysicalGroupLookup
  */
 TupleTableSlot *
 ExecPhysicalGroupLookup(PhysicalGroupLookupState *node)
 {
 	NestLoopState *outer = (NestLoopState *) outerPlanState(node);
-	TupleTableSlot *result = ExecProcNode((PlanState *) outer);
-	TupleTableSlot *inner = outer->js.ps.ps_ExprContext->ecxt_innertuple;
+	TupleTableSlot *result;
+	TupleTableSlot *inner;
+	bool res;
+	EState *estate = outer->js.ps.state;
+	Buffer buffer;
+	HeapUpdateFailureData hufd;
+	HeapTuple tup;
+	Relation rel;
+	TupleTableSlot *slot = NULL;
+	BitmapHeapScanState *scan;
 
+	/* Get next tuple from subplan, if any. */
+lnext:
```

Comment:
That's how Postgres does it in `nodeLockRows.c`. This logic is pretty similar to what goes inside that.